core/domain/feedback_services.py

```diff
@@ -172,3 +188,99 @@ def get_thread_analytics(exploration_id):
     """
     return feedback_jobs.FeedbackAnalyticsAggregator.get_thread_analytics(
         exploration_id)
+
+
+def create_suggestion(exploration_id, author_id, exploration_version,
+                      state_name, suggestion_content):
+    """Creates a new SuggestionModel object and the corresponding
+    FeedbackThreadModel object."""
+
+    thread_id = _create_models_for_thread_and_first_message(
+        exploration_id, state_name, author_id, 
+        DEFAULT_SUGGESTION_THREAD_SUBJECT, EMPTY_TEXT, True)
+    (feedback_models.SuggestionModel.create(
+        exploration_id, thread_id, author_id, exploration_version, state_name,
+        suggestion_content))
+
+
+def get_suggestion(exploration_id, thread_id):
+    return feedback_models.SuggestionModel.get_by_exploration_and_thread_id(
+        exploration_id, thread_id)
+
+
+def get_open_threads(exploration_id, suggestions):
+    """If suggestions is True, return a list of all open threads that have a
+    suggestion, otherwise return a list of all open threads that do not have a 
+    suggestion."""     
+
+    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    open_threads = []
+    for thread in threads:
+        if (bool(thread.has_suggestion) == bool(suggestions) and 
+            thread.status == feedback_models.STATUS_CHOICES_OPEN):
+            open_threads.append(thread)
+    return open_threads
+
+
+def get_closed_threads(exploration_id, suggestions):
+    """If suggestions is True, return a list of all closed threads that have a
+    suggestion, otherwise return a list of all closed threads that do not have a 
+    suggestion."""     
+
+    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    closed_threads = []
+    for thread in threads:
+        if (bool(thread.has_suggestion) == bool(suggestions) and 
+            thread.status != feedback_models.STATUS_CHOICES_OPEN):
+            closed_threads.append(thread)
+    return closed_threads
+
+
+def get_all_suggestion_threads(exploration_id):
+    """Return a list of all threads with suggestions."""
+
+    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
```

Comment:
Umm, I'm not sure how. get_threads will return a list which does not have a filter method. (Unless you're suggestion not using get_threads and using model layer api's directly here).